storagenode/orders/sender.go

```diff
@@ -201,20 +254,21 @@ func (sender *Sender) settle(ctx context.Context, log *zap.Logger, satelliteID s
 			break
 		}
 
+		var status Status
 		switch response.Status {
 		case pb.SettlementResponse_ACCEPTED:
-			err = sender.orders.Archive(ctx, satelliteID, response.SerialNumber, StatusAccepted)
-			if err != nil {
-				errHandle(OrderError, "failed to archive order as accepted: serial: %v, %v", response.SerialNumber, err)
-			}
+			status = StatusAccepted
 		case pb.SettlementResponse_REJECTED:
-			err = sender.orders.Archive(ctx, satelliteID, response.SerialNumber, StatusRejected)
-			if err != nil {
-				errHandle(OrderError, "failed to archive order as rejected: serial: %v, %v", response.SerialNumber, err)
-			}
+			status = StatusRejected
 		default:
 			errHandle(OrderError, "unexpected response: %v", response.Status)
 		}
+
+		ch <- ArchiveRequest{
```

Comment:
won't order be archived even with `unexpected response`?

Reply:
indeed. fixing.